OrbitClientModel/include/OrbitClientModel/CaptureSerializer.h

```diff
@@ -0,0 +1,87 @@
+// Copyright (c) 2020 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef ORBIT_CLIENT_MODEL_CAPTURE_SERIALIZER_H_
+#define ORBIT_CLIENT_MODEL_CAPTURE_SERIALIZER_H_
+
+#include <iosfwd>
+#include <outcome.hpp>
+#include <string>
+
+#include "CaptureData.h"
+#include "OrbitBase/Result.h"
+#include "capture_data.pb.h"
+#include "google/protobuf/io/coded_stream.h"
+#include "google/protobuf/io/zero_copy_stream_impl.h"
+#include "google/protobuf/message.h"
+
+namespace CaptureSerializer {
+
+template <class TimersIterator>
+ErrorMessageOr<void> Save(const std::string& filename, const CaptureData& capture_data,
+                          const absl::flat_hash_map<uint64_t, std::string>& key_to_string_map,
+                          TimersIterator timers_iterator_begin, TimersIterator timers_iterator_end);
+
+void WriteMessage(const google::protobuf::Message* message,
+                  google::protobuf::io::CodedOutputStream* output);
+
+}  // namespace CaptureSerializer
+
+namespace internal {
+
+template <class TimersIterator>
+void Save(std::ostream& stream, const CaptureData& capture_data,
+          const absl::flat_hash_map<uint64_t, std::string>& key_to_string_map,
+          TimersIterator timers_iterator_begin, TimersIterator timers_iterator_end);
+
+orbit_client_protos::CaptureInfo GenerateCaptureInfo(
+    const CaptureData& capture_data,
+    const absl::flat_hash_map<uint64_t, std::string>& key_to_string_map);
+
+inline const std::string kRequiredCaptureVersion = "1.52";
+
+}  // namespace internal
+
+template <class TimersIterator>
+void internal::Save(std::ostream& stream, const CaptureData& capture_data,
+                    const absl::flat_hash_map<uint64_t, std::string>& key_to_string_map,
+                    TimersIterator timers_iterator_begin, TimersIterator timers_iterator_end) {
+  google::protobuf::io::OstreamOutputStream out_stream(&stream);
+  google::protobuf::io::CodedOutputStream coded_output(&out_stream);
+
+  orbit_client_protos::CaptureHeader header;
+  header.set_version(kRequiredCaptureVersion);
```

Comment:
I have a question here which I just realise: is it needed to use `internal::kRequiredCaptureVersion` here instead? If not, why? If it does, why does the code compile and works? Thanks!

Reply:
You don't need the `internal::` because it's in the same namespace as the function since `Save` is also part of `internal`.

Reply:
Ok, I understand, thanks a lot!